src/main/java/com/google/sps/servlets/BlobstoreUploadUrlServlet.java

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+ 
+package com.google.sps.servlets;
+ 
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.sps.utils.SafeSearchUtils;
+ 
+/**
+ * When the fetch() function requests the /blobstore-upload-url URL, the content of the response is
+ * the URL that allows a user to upload a file to Blobstore. 
+ * The user's browser uploads the file directly to the Blobstore via the generated URL.
+ * Blobstore rewrites the request to contain a blobkey 
+ */
+@WebServlet("/blobstore-upload-url")
+public class BlobstoreUploadUrlServlet extends HttpServlet {
+ 
+  BlobstoreService blobstoreService;
+ 
+  public BlobstoreUploadUrlServlet(){
+    blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+  }
+ 
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String uploadUrl = blobstoreService.createUploadUrl("/image-upload-handler-blobstore");
+
+    // TODO: scan for inappropriate content: SafeSearchUtils.detectSafeSearch(uploadUrl);
```

Comment:
I believe the place to implement this TODO is in the /image-upload-handler-blobstore servlet. This /blobstore-upload-url servlet doesn't have access to the image, which hasn't been uploaded yet.

Reply:
Alright, I moved the TODO there.